docs/how-to-setup-wsl.md

```diff
@@ -0,0 +1,218 @@
+# Setting up WSL for Working with freeCodeCamp Locally
+
+Follow these guidlines for setting up WSL to set up freeCodeCamp locally.
+
+This guide is covers some common issues encountered with the setup of WSL.
+
+## Enable WSL & Download a Distro
+
+1) Follow the instructions given in the documentation by Microsoft to install WSL2 (Recommended Distro: Ubuntu18.04):
+[Installing WSL2](https://docs.microsoft.com/en-us/windows/nodejs/setup-on-wsl2#install-windows-10-insider-preview-build)
+
+> [!NOTE]
+> You can download a different distro, but the setup might differ.
+
+2) On the same page, ensure to follow the instructions about installing Nodejs and NPM through the NVM.
+
+**NOTE:** If you encounter errors on step 2, skip to the next step, and return when prompted
+
+## Setting Up Git
+
+It is likely you will be unable to clone your fork of freeCodeCamp, without running these steps, but there is no harm in trying. Just follow the instructions on [the Contributors Site](https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally).
+
+If you run into any issues, follow through with one alternative in this step:
+
+### ALTERNATIVE 1 - Use apt-get to Install the Latest Version of Git
+
+1) Update your distrobution:
+
+```sh
+sudo apt update && sudo apt upgrade
+```
+
+2) Download and install Git:
+
+```sh
+sudo apt install git
+```
+
+3) Verify the installed version:
+
+```sh
+git --version
+```
+
+### ALTERNATIVE 2 - Use SSH to Work on Your Fork
+
+1) Follow the instructions on [GitHub](https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh) to generate a new SSH key, and add it to your account.
+
+2) If you encounter an error similar to this, perform the rest of this alternative:
+
+ >ssh: connect to host `github.com` port 22: Connection refused
+
+3) CD into the `.ssh` folder, and create a config file:
+
+```sh
+cd ~/.ssh && touch config
+```
+
+4) Open the file with your favourite text editor:
+
+Open VS Code: `code .`
+
+Alternatively, open the folder in the explorer, and use the Windows built-in text editor: `explorer.exe .`
+
+> [!NOTE]
+> This command works for VS Code, and you might be asked to install an extension for this to work
+
+5) Add this to the config file to allow GitHub's domain through your firewall:
+
+```
+# GitHub Account
+Host github.com
+    HostName ssh.github.com
+    Port 443
+    PreferredAuthentications publickey
+    IdentityFile ./id_rsa.pub
+```
+
+> [!DANGER]
+> Provided you have not deviated from the default setup, the `IdentityFile` will be as above.
+
+6) Return to the GitHub SSH setup to test your connection
+
+### ALTERNATIVE 3 - Last Resort
```

Comment:
If we are asking users to do the steps in this part, we are expecting a lot from them. Meaning, we are maintainers have gotten this wrong and need to provide a better way.